### PR TITLE
Move to scss on the DEV branch

### PR DESCRIPTION
**Pull request by elessar-ch**
This PR includes all the changes by @sventschui to move to scss.

### CODE REVIEW

**Comment by sventschui** [comment]
Guys, can we merge this into develop please? Some projects starting soon in axa that would love to use scss. I'd migrate myaxa. Thanks!

**Comment by davidknezic** [comment]
Yup, that's even better! So we don't drift away too much!

**Comment by elessar-ch** [comment]
@sventschui Happily

**Comment by AndyOGo** [comment]
I just recently fixed the compatibility of [HappyPack and ExtraxtTextWebpackPlugin + PostCSSLoader](https://github.com/axa-ch/style-guide/pull/618)
Depending on which gets merged first, the will have to fix conflicts;)

**Comment by elessar-ch** [comment]
That's not funny @AndyOGo 😡

**Comment by elessar-ch** [comment]
So, I think we can merge this now

**Comment by elessar-ch** [comment]
I'll then have a look at trying to merge it into @AndyOGo's branch

**Comment by AndyOGo** [comment]
@elessar-ch 
👍 If you need something, feel free to ask

**Comment by elessar-ch** [comment]
@davidknezic @sventschui @AndyOGo Merge?

File: scss/style/blocks/dialog.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import '../mixins/respond';
+@import './modal';
```
Comment (AndyOGo) [inline scss/style/blocks/dialog.scss]: That should be `modal2` or?

File: scss/style/blocks/dropdown-2.scss [hunk 2]
```diff
@@ -1,33 +1,33 @@
-@import '../mixins/clearfix.less';
-@import '../mixins/font.less';
-@import '../mixins/make-link.less';
-@import '../mixins/target.less';
+@import '../mixins/clearfix.scss';
+@import '../mixins/font.scss';
+@import '../mixins/make-link.scss';
+@import '../mixins/target.scss';
 @import '../variables';
```
Comment (AndyOGo) [inline scss/style/blocks/dropdown-2.scss]: missing `.scss` suffix

File: scss/style/blocks/dropzone.scss [hunk 3]
```diff
@@ -2,16 +2,16 @@
 @import '../variables';
```
Comment (AndyOGo) [inline scss/style/blocks/dropzone.scss]: missing `.scss` suffix

File: scss/style/blocks/footer-copyline.scss [hunk 4]
```diff
@@ -3,32 +3,32 @@
 @import '../mixins/unstyle-list';
```
Comment (AndyOGo) [inline scss/style/blocks/footer-copyline.scss]: missing `.scss` suffix

File: scss/style/blocks/form.scss [hunk 5]
```diff
@@ -1,19 +1,19 @@
 @import '../mixins/respond';
```
Comment (AndyOGo) [inline scss/style/blocks/form.scss]: missing `.scss` suffix

File: scss/style/blocks/header-search.scss [hunk 6]
```diff
@@ -4,24 +4,24 @@
 @import './control';
```
Comment (AndyOGo) [inline scss/style/blocks/header-search.scss]: missing `.scss` suffix

File: scss/style/blocks/header.scss [hunk 7]
```diff
@@ -7,42 +7,42 @@
 @import '../mixins/respond';
```
Comment (AndyOGo) [inline scss/style/blocks/header.scss]: missing `.scss` suffix

File: scss/style/blocks/icon-teaser.scss [hunk 8]
```diff
@@ -10,25 +10,25 @@
 @import './icon';
```
Comment (AndyOGo) [inline scss/style/blocks/icon-teaser.scss]: missing `.scss` suffix

File: scss/style/blocks/main-nav.scss [hunk 9]
```diff
@@ -2,9 +2,9 @@
 @import '../mixins/respond';
```
Comment (AndyOGo) [inline scss/style/blocks/main-nav.scss]: missing `.scss` suffix

File: scss/style/blocks/menu.scss [hunk 10]
```diff
@@ -5,7 +5,7 @@
 @import '../mixins/word-break';
```
Comment (AndyOGo) [inline scss/style/blocks/menu.scss]: missing `.scss` suffix

File: scss/style/mixins/ellipsis.scss [hunk 11]
```diff
@@ -0,0 +1,12 @@
+// $see https://css-tricks.com/snippets/css/prevent-long-urls-from-breaking-out-of-container/
+@mixin ellipsis($textOverflow: 'ellipsis') {
+  @if $textOverflow == 'ellipsis' {
```
Comment (AndyOGo) [inline scss/style/mixins/ellipsis.scss]: this is wrong, it should check for `'none'` and only the reset the ellipsis styles

File: webpack/bundle.config.babel.js [hunk 12]
```diff
@@ -41,14 +40,14 @@ export default {
       exclude: /node_modules/,
       loader: `happypack/loader?id=${getEnvId('jsx')}`,
     }, {
-      test: /\.less/,
+      test: /\.scss$/,
       loader: ExtractTextPlugin.extract('style', [
         'css?importLoaders=2&sourceMap',
         'postcss-loader',
-        'less?outputStyle=expanded&sourceMap=true&sourceMapContents=true',
+        'sass',
```
Comment (AndyOGo) [inline webpack/bundle.config.babel.js]: Do we not want to pass some params to sass here, similiar to `?outputStyle=expanded&sourceMap=true&sourceMapContents=true`

**Comment by AndyOGo** [comment]
upps, I think I was too late 😞

**Comment by elessar-ch** [comment]
I favor not using the extension.
So for the ellipsis thing it should be `@if $textOverflow == 'ellipsis' {`?
Yeah, the options for the sass loader should probably be added for debugging especially.

**Comment by elessar-ch** [comment]
We'll just fix it on develop :P

File: webpack/bundle.config.babel.js (hunk 12 above)
Comment (elessar-ch) [inline webpack/bundle.config.babel.js]: Ahh... The options are there in the `bundle.dev.config.babel.js`. I think for prod this would be ok?

**Comment by AndyOGo** [comment]
@elessar-ch 
no, i fixed it already on develop and pushed - it should be:

````scss
@mixin ellipsis($textOverflow: 'ellipsis') {
  @if $textOverflow == 'none' {
    overflow: visible;
    white-space: normal;
    text-overflow: clip;
  } @else {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: unquote($textOverflow);
  }
}
````

**Comment by AndyOGo** [comment]
and i fixed the modal2 dependency in the dialog

**Comment by AndyOGo** [comment]
@elessar-ch 
okidok, so lets remove all extension, i will fix

**Comment by elessar-ch** [comment]
Nono, I'm on it

**Comment by elessar-ch** [comment]
@AndyOGo u sure about that modal vs modal2?

I get the following error message now:
```
[all:watch] [bundles:watch] ERROR in ./~/css-loader?importLoaders=2&sourceMap!./~/postcss-loader!./~/sass-loader?outputStyle=expanded&sourceMap=true&sourceMapContents=true!./docs/scss/docs.scss
[all:watch] [bundles:watch] Module build failed:
[all:watch] [bundles:watch]   padding: $modal-padding;
[all:watch] [bundles:watch]           ^
[all:watch] [bundles:watch]       Undefined variable: "$modal-padding".
[all:watch] [bundles:watch]       in C:\dev\repos\axa-ch\style-guide\scss\style\blocks\dialog.scss (line 99, column 12)
[all:watch] [bundles:watch]  @ ./docs/scss/docs.scss 4:14-251 13:2-17:4 14:20-257
```

**Pull request merged**